src/PDE_BPINN.jl

```diff
@@ -0,0 +1,502 @@
+mutable struct PDELogTargetDensity{
+    ST <: AbstractTrainingStrategy,
+    D <: Union{Nothing, Vector{<:Matrix{<:Real}}},
+    P <: Vector{<:Distribution},
+    I,
+    F,
+    PH,
+}
+    dim::Int64
+    strategy::ST
+    dataset::D
+    priors::P
+    allstd::Vector{Vector{Float64}}
+    names::Tuple
+    extraparams::Int
+    init_params::I
+    full_loglikelihood::F
+    Φ::PH
+
+    function PDELogTargetDensity(dim, strategy, dataset,
+            priors, allstd, names, extraparams,
+            init_params::AbstractVector, full_loglikelihood, Φ)
+        new{
+            typeof(strategy),
+            typeof(dataset),
+            typeof(priors),
+            typeof(init_params),
+            typeof(full_loglikelihood),
+            typeof(Φ),
+        }(dim,
+            strategy,
+            dataset,
+            priors,
+            allstd,
+            names,
+            extraparams,
+            init_params,
+            full_loglikelihood,
+            Φ)
+    end
+    function PDELogTargetDensity(dim, strategy, dataset,
+            priors, allstd, names, extraparams,
+            init_params::Union{NamedTuple, ComponentArrays.ComponentVector},
+            full_loglikelihood, Φ)
+        new{
+            typeof(strategy),
+            typeof(dataset),
+            typeof(priors),
+            typeof(init_params),
+            typeof(full_loglikelihood),
+            typeof(Φ),
+        }(dim,
+            strategy,
+            dataset,
+            priors,
+            allstd,
+            names,
+            extraparams,
+            init_params,
+            full_loglikelihood,
+            Φ)
+    end
+end
+
+function LogDensityProblems.logdensity(Tar::PDELogTargetDensity, θ)
+    # for parameter estimation neccesarry to use multioutput case
+    return Tar.full_loglikelihood(setparameters(Tar, θ),
+               Tar.allstd) + priorlogpdf(Tar, θ) + L2LossData(Tar, θ)
+    # + L2loss2(Tar, θ)
+end
+
+# function L2loss2(Tar::PDELogTargetDensity, θ)
+#     return Tar.full_loglikelihood(setparameters(Tar, θ),
+#         Tar.allstd)
+# end
+
+function setparameters(Tar::PDELogTargetDensity, θ)
```

Comment:
Doesn't lux have a utility for the transformation of vector into componentvector for this kind of thing?

Reply:
i couldnt find it in the docs(atleaast something specific to this case), so i worked it out

Reply:
@avik-pal

Reply:
Let's make sure an issue get's opened about this after merging.

Reply:
Typically you store the axes of the ComponentVector via `ax = getaxes(ps_ca)` and then construct the CA via `ComponentArray(theta, ax)`